datasafe-long-run-tests/datasafe-business-tests-random-actions/src/test/java/de/adorsys/datasafe/business/impl/e2e/randomactions/framework/BaseRandomActions.java

```diff
@@ -78,12 +79,57 @@ protected Fixture fixture(String path) {
     @ValueSource
     protected static Stream<Arguments> actionsOnSoragesAndThreadsAndFilesizes() {
         return Sets.cartesianProduct(
-                Collections.singleton(minio()),
+                Collections.singleton(s3()),
             THREAD_COUNT,
             FILE_SIZE_M_BYTES
         ).stream().map(it -> Arguments.of(it.get(0), it.get(1), it.get(2)));
     }
 
+    @ValueSource
+    protected static Stream<Arguments> actionsOnMultiStorageParallelThreads() {
+        List<String> users = new ArrayList<>();
+        smallFixture().getOperations().stream().forEach((i)->users.add(i.getUserId()));
+
+        return Sets.cartesianProduct(
+                Collections.singleton(getMultiS3Bucket(users)),
+                THREAD_COUNT,
+                FILE_SIZE_M_BYTES
+        ).stream().map(it -> Arguments.of(it.get(0), it.get(1), it.get(2)));
+    }
+
+    protected void executeMultiStorageTest(Fixture fixture,
```

Comment:
There is no need of this duplicate function,
createUsersMultiStorage does nothing more than createUsers does

Reply:
All the functionalities are required except from queue and log why because, Operation Executor changed to different for multi bucket process(to identify the user exist/selected bucket). And create users method is also uses the same.

Reply:
It is not needed. You need to provide storage service (StorageService) implementation that knows which user uses which AmazonS3